patches/gcc-10.1.0/0004-libgomp-test-fix.diff

```diff
@@ -0,0 +1,62 @@
+diff --git a/libgomp/testsuite/Makefile.am b/libgomp/testsuite/Makefile.am
+index 655a413c160..916e9d30f15 100644
+--- a/libgomp/testsuite/Makefile.am
++++ b/libgomp/testsuite/Makefile.am
+@@ -14,6 +14,13 @@ RUNTESTDEFAULTFLAGS = --tool $$tool --srcdir $$srcdir
+ 
+ EXTRA_DEJAGNU_SITE_CONFIG = libgomp-site-extra.exp
+ 
++EXTRA_DEJAGNU_SITE_CONFIG = extra.exp
++
++extra.exp:
++	echo 'set BUILD_CC "$(CC)"' > $@.tmp
++	mv $@.tmp $@
++
++
+ # Instead of directly in ../testsuite/libgomp-test-support.exp.in, the
+ # following variables have to be "routed through" this Makefile, for expansion
+ # of the several (Makefile) variables used therein.
+diff --git a/libgomp/testsuite/Makefile.in b/libgomp/testsuite/Makefile.in
+index 52aa6c5fbc9..617da8ddbf2 100644
+--- a/libgomp/testsuite/Makefile.in
++++ b/libgomp/testsuite/Makefile.in
+@@ -311,6 +311,7 @@ _RUNTEST = $(shell if test -f $(top_srcdir)/../dejagnu/runtest; then \
+ 	     echo $(top_srcdir)/../dejagnu/runtest; else echo runtest; fi)
+ 
+ RUNTESTDEFAULTFLAGS = --tool $$tool --srcdir $$srcdir
++EXTRA_DEJAGNU_SITE_CONFIG = extra.exp
+ EXTRA_DEJAGNU_SITE_CONFIG = libgomp-site-extra.exp
+ all: all-am
+ 
+@@ -475,6 +476,10 @@ uninstall-am:
+ .PRECIOUS: Makefile
+ 
+ 
++extra.exp:
++	echo 'set BUILD_CC "$(CC)"' > $@.tmp
++	mv $@.tmp $@
++
+ # Instead of directly in ../testsuite/libgomp-test-support.exp.in, the
+ # following variables have to be "routed through" this Makefile, for expansion
+ # of the several (Makefile) variables used therein.
+diff --git a/libgomp/testsuite/lib/libgomp.exp b/libgomp/testsuite/lib/libgomp.exp
+index ee5f0e57b19..183081b5802 100644
+--- a/libgomp/testsuite/lib/libgomp.exp
++++ b/libgomp/testsuite/lib/libgomp.exp
+@@ -68,6 +68,7 @@ proc libgomp_init { args } {
+     global ALWAYS_CFLAGS
+     global CFLAGS
+     global TOOL_EXECUTABLE TOOL_OPTIONS
++    global BUILD_CC
+     global GCC_UNDER_TEST
+     global TESTING_IN_BUILD_TREE
+     global target_triplet
+@@ -90,6 +91,8 @@ proc libgomp_init { args } {
+     if ![info exists GCC_UNDER_TEST] then {
+ 	if [info exists TOOL_EXECUTABLE] {
+ 	    set GCC_UNDER_TEST $TOOL_EXECUTABLE
++	} elseif [info exists BUILD_CC] {
++	    set GCC_UNDER_TEST $BUILD_CC        
+ 	} else {
+ 	    set GCC_UNDER_TEST "[find_gcc]"
+ 	}
```

Comment:
This might have been [fixed](https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=c8e759b4215ba4b376c9d468aeffe163b3d520f0) [upstream](https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=749bd22ddc50b5112e5ed506ffef7249bf8e6fb3) in another way, as well?

Reply:
You are right. I am generating patches for gcc10.2.0 anyway, I will update this PR

Reply:
Thank you for your revision